app/models/tslr_claim.rb

```diff
@@ -142,4 +143,26 @@ def ni_number_is_correct_format
     errors.add(:national_insurance_number, "Enter a National Insurance number in the correct format") \
       if national_insurance_number.present? && !normalised_ni_number.match(/\A[a-z]{2}[0-9]{6}[a-d]{1}\Z/i)
   end
+
+  def normalise_account_number
+    self.account_number = normalised_bank_detail(account_number)
+  end
+
+  def normalise_account_sort_code
+    self.account_sort_code = normalised_bank_detail(account_sort_code)
+  end
+
+  def normalised_bank_detail(bank_detail)
+    bank_detail.gsub(/\D/, "")
```

Comment:
This is effectively doing the same normalisation as the TRN normalisation. Is there a way to combine the two methods into a single one that normalises a numeric value?

Thinking about this a bit more, I think in reality we could have probably been more specific with the normalisation (of the TRN and these fields) and only strip characters that we think users would legitimately enter, i.e. whitespace and '-', and then validate what they've entered is actually positive numeric integer (i.e. `/\A\d+\z/`). Otherwise the normalisation means that a user could enter `"ABC1234XX?12"` and we'd accept that as a sort code, normalising it to `123412`. I don't think there's a huge risk or danger here but something to keep in mind.

Reply:
I've had a second attempt at this, we should validate againse more now as per your example, see what you think?